src/store.cpp

```diff
@@ -1,92 +1,82 @@
-    command.unit = doc["unit"].as<std::string>();
-    command.active = doc["active"].as<bool>();
-    command.interval = doc["interval"].as<uint32_t>();
-    command.last = 0;
-    command.master = doc["master"].as<bool>();
-    command.position = doc["position"].as<size_t>();
-    command.datatype =
-        ebus::string2datatype(doc["datatype"].as<const char *>());
-    command.topic = doc["topic"].as<std::string>();
-    command.ha = doc["ha"].as<bool>();
-    command.ha_class = doc["ha_class"].as<std::string>();
-
-    const std::vector<Command>::const_iterator it =
-        std::find_if(allCommands.begin(), allCommands.end(),
-                     [&key](const Command &cmd) { return cmd.key == key; });
-
-    if (it != allCommands.end()) allCommands.erase(it);
-
-    allCommands.push_back(command);
-    countCommands();
-    publishCommand(&allCommands.back(), false);
-    if (command.ha) publishHomeAssistant(&allCommands.back(), false);
+void Store::publishCommands() {
+  for (const Command &command : allCommands) pubCommands.push_back(&command);
 
-    lastInsert = millis();
-  }
+  if (allCommands.size() == 0) mqtt.publish("commands", 0, false, "");
 }
 
-void Store::removeCommand(const char *payload) {
-  JsonDocument doc;
-  DeserializationError error = deserializeJson(doc, payload);
+void Store::loadCommands() {
+  Preferences commands;
+  commands.begin("commands", true);
 
-  if (error) {
-    std::string err = "DeserializationError ";
-    err += error.c_str();
-    mqttClient.publish("ebus/config/error", 0, false, err.c_str());
+  size_t bytes = commands.getBytesLength("ebus");
+  if (bytes > 2) {  // 2 = empty json array "[]"
+    std::vector<char> buffer(bytes);
+    bytes = commands.getBytes("ebus", buffer.data(), bytes);
+    if (bytes > 2) {
+      std::string payload(buffer.begin(), buffer.end());
+      deserializeCommands(payload.c_str());
+      publishResponse("load", "successful", bytes);
+    } else {
+      publishResponse("load", "failed");
+    }
   } else {
-    std::string key = doc["key"].as<std::string>();
+    publishResponse("load", "no data");
+  }
 
-    const std::vector<Command>::const_iterator it =
-        std::find_if(allCommands.begin(), allCommands.end(),
-                     [&key](const Command &cmd) { return cmd.key == key; });
+  commands.end();
+}
 
-    if (it != allCommands.end()) {
-      publishCommand(&(*it), true);
-      if (it->ha) publishHomeAssistant(&(*it), true);
-      allCommands.erase(it);
-      countCommands();
-    } else {
-      std::string err = key + " not found";
-      mqttClient.publish("ebus/config/error", 0, false, err.c_str());
-    }
+void Store::saveCommands() const {
+  Preferences commands;
+  commands.begin("commands", false);
+
+  size_t bytes = strlen(serializeCommands().c_str());
+  if (bytes > 2) {  // 2 = empty json array "[]"
+    bytes = commands.putBytes("ebus", serializeCommands().c_str(), bytes);
```

Comment:
have you tested this with some bigger amount of commands (several kBs)? no memory issues?

Reply:
okay, we will see...